components/admin/lessons/AdminLessonInfo.tsx

```diff
@@ -147,15 +148,46 @@ export const AdminLessonInfo: React.FC<LessonInfoProps> = ({
   lessons,
   selectedLesson
 }) => {
+  const [newChallengeView, setNewChallengeView] = useState(false)
+
   // true when user clicks on `create new lesson` button
   if (lessons && selectedLesson === lessons.length - 1) {
     return <NewLesson setLessons={setLessons} />
   }
+
   // set currently selected lesson
   const lesson = lessons && lessons[selectedLesson]
+
+  const lessonId = parseInt(lesson ? lesson.id + '' : '')
+
+  const newChallengeClick = () => setNewChallengeView(!newChallengeView)
+
+  const viewOption = newChallengeView ? (
+    <NewChallenge setLessons={setLessons} lessonId={lessonId} />
+  ) : (
+    <>
```

Comment:
What do empty component tags resolve to?

Reply:
They are called fragments https://reactjs.org/docs/fragments.html
It is so that I don't have to use a `div` html element to wrap list of children

Reply:
Ah, ok. I hadn't seen the short syntax before.

Reply:
@anthonykhoa I thought they need to be self closing? `< />`

Reply:
The closing tag is on line 180.